data-prepper-core/src/test/java/org/opensearch/dataprepper/peerforwarder/discovery/DnsPeerListProviderTest.java

```diff
@@ -90,17 +90,27 @@ public void testGetPeerList() {
     }
 
     @Test
-    public void testActivePeerCounter() {
+    public void testActivePeerCounter_with_list() {
         when(dnsAddressEndpointGroup.endpoints()).thenReturn(ENDPOINT_LIST);
 
-        final List<Measurement> endpointsMeasures = MetricsTestUtil.getMeasurementList(new StringJoiner(MetricNames.DELIMITER).add(COMPONENT_SCOPE).add(COMPONENT_ID)
-                .add(PeerListProvider.PEER_ENDPOINTS).toString());
-        assertEquals(1, endpointsMeasures.size());
-        final Measurement endpointsMeasure = endpointsMeasures.get(0);
-        assertEquals(2.0, endpointsMeasure.getValue(), 0);
+        final ArgumentCaptor<ToDoubleFunction<DnsAddressEndpointGroup>> gaugeFunctionCaptor = ArgumentCaptor.forClass(ToDoubleFunction.class);
+        verify(pluginMetrics).gauge(eq(PEER_ENDPOINTS), eq(dnsAddressEndpointGroup), gaugeFunctionCaptor.capture());
+
+        final ToDoubleFunction<DnsAddressEndpointGroup> gaugeFunction = gaugeFunctionCaptor.getValue();
 
+        assertThat(gaugeFunction.applyAsDouble(dnsAddressEndpointGroup), equalTo(2.0));
+    }
+
+    @Test
+    public void testActivePeerCounter_with_single() {
         when(dnsAddressEndpointGroup.endpoints()).thenReturn(Collections.singletonList(Endpoint.of(ENDPOINT_1)));
-        assertEquals(1.0, endpointsMeasure.getValue(), 0);
+
+        final ArgumentCaptor<ToDoubleFunction<DnsAddressEndpointGroup>> gaugeFunctionCaptor = ArgumentCaptor.forClass(ToDoubleFunction.class);
+        verify(pluginMetrics).gauge(eq(PEER_ENDPOINTS), eq(dnsAddressEndpointGroup), gaugeFunctionCaptor.capture());
+
+        final ToDoubleFunction<DnsAddressEndpointGroup> gaugeFunction = gaugeFunctionCaptor.getValue();
+
+        assertThat(gaugeFunction.applyAsDouble(dnsAddressEndpointGroup), equalTo(1.0));
```

Comment:
This looks somewhat strange. No call to `pluginMetrics.gauge(...)` is visible. For me, this makes the test a little harder to understand.